### ISSUE TITLE
Packaging with vcpkg

### ISSUE THREAD

**Issue by ekilmer** [cginternals/cpplocate#47]
I just thought I'd let it be known that I have a draft PR for packaging this project in vcpkg here https://github.com/microsoft/vcpkg/pull/23173

I want to give a few days for the maintainer(s) of this project to look it over and make comments if they'd like.

I had to make a few small patches:

1. Detection of `SYSTEM_INSTALL_DIR` doesn't work very well when using vcpkg (they set the `CMAKE_INSTALL_PREFIX` to their staging area, which is different than `/usr` and `/usr/local`) and there is no way to override the installation paths.
  
    * Possible fix: Use CMake [GNUInstallDirs](https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html) for the install directory logic and remove the `SYSTEM_INSTALL_DIR` stuff. (Looks like a PR trying to address this is still open https://github.com/cginternals/cpplocate/pull/15, but I think use of `GNUInstallDirs` is probably the best path)

2. Very small changes to the config files that are the result of vcpkg's own fixing/moving of the export files to support both release and debug builds. I also had to move `liblocate-config.cmake` into its own directory next to the moved/fixed liblocate CMake export files. This appears in `portfile.cmake`.

    * Not sure if there is a good fix to upstream here

~There is a failure to build on Windows UWP. I'm not very familiar with [UWP](https://docs.microsoft.com/en-us/windows/uwp/get-started/universal-application-platform-guide) and its differences, but I've attached the error log here: [install-x64-uwp-dbg-out.log](https://github.com/cginternals/cpplocate/files/8100719/install-x64-uwp-dbg-out.log).~ ~Fixed by replacing all `SYSTEM_WINDOWS` define checks with just `_WINDOWS`. `SYSTEM_WINDOWS` is coming from this project's `cmake/CompileOptions.cmake` and is defined as `SYSTEM_WINDOWSSTORE` during UWP builds, but these are not checked, whereas `_WINDOWS` is coming internally from CMake... I'm not sure what is more correct: adding another check for `SYSTEM_WINDOWSSTORE` or relying on `_WINDOWS`.~ Nevermind, I now decided that it's probably better to just indicate that UWP isn't supported. If anyone knows more about this, I'm happy to make changes.

Let me know if I can be of any help or you'd like to further discuss the vcpkg PR.

Thank you for the great library!

**Comment by scheibel** [comment]
Thanks for the summary and your commitment to integrate this library into vcpkg!

We will definitively have a look at `GNUInstallDirs`, this is not the first time this module is suggested to us.

Regarding your proposed changes, I'm still collecting the different use cases and requirements from the different packaging solutions on all those platforms we want our libraries to be available on. Your changes will be part of our analysis when we find time for a rework of the install configuration.

But let me explain why we don't integrate deployment changes - although really valuable for the vcpkg community - into our libraries right away:
Adapting our default setup to one package manager, such as vcpkg, is not a hard task, but it mainly will be incompatible with the expectations from other packaging solutions, such as Debian packages, PPAs, AUR, macports, Homebrew, and even a package-manager-free install from source.
For now, we decided to adopt small, unintrusive changes that hopefully won't break other hand-crafted packaging solutions such as yours but for all other package managers our libraries are already available on.
In short, we expect that the packaging of our libraries will be incompatible in parts or even in full with the requirements of packaging solutions, such that each package maintainer will adopt or even write whole new packaging files.

**Comment by ekilmer** [comment]
> We will definitively have a look at GNUInstallDirs, this is not the first time this module is suggested to us.
>
> Regarding your proposed changes, I'm still collecting the different use cases and requirements from the different packaging solutions on all those platforms we want our libraries to be available on. Your changes will be part of our analysis when we find time for a rework of the install configuration.

Thank you! Let me know if you would like to discuss purpose or reason for the changes further.

------------------

> In short, we expect that the packaging of our libraries will be incompatible in parts or even in full with the requirements of packaging solutions, such that each package maintainer will adopt or even write whole new packaging files.

Of course! Don't get me wrong, the CMake and packaging of this project is quite good in comparison to many other project's, especially considering this project maintains compatibility with CMake 3.0! 

I do not expect you to integrate the change(s) to the package config files, especially if only vcpkg is requesting the changes, so that's okay.

The use of `GNUInstallDirs`, which you have had previous requests, is something that I think could really help most packaging situations because of that module's use of CMake CACHE variables, which allow packagers to customize the options without patching. Of course, even changing installation directory layouts can have unforeseen consequences, which is actually a big issue for people using this project to find files with relative paths!!!

Thank you again for your responses and consideration 😄 

Feel free to close this issue whenever you'd like, else I will close it when the vcpkg PR is merged 👍
